### ISSUE TITLE
Investigate supporting DI on deserialization via MobileFormatter

### ISSUE THREAD

**Issue by rockfordlhotka** [MarimerLLC/csla#1044]
It appears that Newtonsoft.Json has the ability to do some sort of ctor DI on deserialization. Something to look into for MobileFormatter perhaps.

https://www.newtonsoft.com/json/help/html/DeserializeWithDependencyInjection.htm

**Comment by Hadronicus** [comment]
Is the idea behind this research to find a way to remove the requirement to have a parameterless constructor for CSLA business objects?

I currently have a working implementation of ICslaReader and ICslaWriter that uses Newtonsoft.Json to serializer and deserialize CSLA objects as JSON using the MobileFormatter. I don’t think it would be too far of a stretch to support importing a [JsonSerializerSetting](https://www.newtonsoft.com/json/help/html/T_Newtonsoft_Json_JsonSerializerSettings.htm) object through the ConfigurationManager to handle things like ContractResolvers, JsonConverters, etc.

**Comment by rockfordlhotka** [comment]
That's not a goal @Hadronicus.

In fact, this was just a discussion I was having with @JasonBock a while back, and thought I'd add a backlog item to remember to research the idea.

Really it might be a problem in search of a solution. As per #787 it is clear that this sort of DI would not be useful for server-side types like database connections. In fact it is hard to imagine it would be useful for _anything_ that would inject non-serializable objects (things that aren't part of the object graph).

At the same time, it is hard to imagine it would be useful for _anything_ that is **serializable** either, because such things would already be part of the object graph.

In other words, this was an idle thought I wanted to consider further, and I have yet to find a problem that it would solve.

Very open to hearing thoughts on what problems it _would_ solve!

**Comment by dazinator** [comment]
If CSLA server side data portal created a DI scope for an incoming request and disposed of that scope after returning a response the I can see value in this. For example an EF Core DbContext registered as a scoped service could be reused for the duration of that request without having to do any special lifetime management anywhere inside of dal fetch methods etc. Just inject the available scoped DbContext instance. It will then be disposed of for you once the server side processing is finished.

**Comment by rockfordlhotka** [comment]
Even with that scenario, @dazinator, I am somewhat skeptical. CSLA 5 does DI on your data portal operation methods, which is a simpler solution.

Why get the dbcontext or DAL reference at a _class level_ when you only need it in your data portal operation method (create/fetch/etc.)? That forces you to store the value in a class-level field, in a non-serializable and not-undoable manner. Seems like more work for no value to me.

**Comment by dazinator** [comment]
@rockfordlhotka ah yes sorry I mis-thought about this. I was thinking purely for injection into at dp operation methods but that isnt what this is. I agree with your analysis - struggling to think of how this would be helpful.

**Comment by rockfordlhotka** [comment]
I am closing this.

In version 3.x we added DI on a per-method basis for data portal operations. To date no one has proposed a specific scenario where DI is necessary for mobile domain objects, so this work item is not needed.
